src/web/source.rs

```diff
@@ -126,19 +91,15 @@ impl FileList {
                         let path_splited: Vec<&str> = path.split('/').collect();
 
                         // if path have '/' it is a directory
-                        let ftype = if path_splited.len() > 1 {
-                            FileType::Dir
-                        } else if mime.starts_with("text") && path_splited[0].ends_with(".rs") {
-                            FileType::RustSource
-                        } else if mime.starts_with("text") {
-                            FileType::Text
+                        let mime = if path_splited.len() > 1 {
+                            "dir".to_owned()
                         } else {
-                            FileType::Binary
+                            mime.to_owned()
                         };
```

Comment:
Why did this change? IIRC we have some custom logic later that does different things for rust files than for anything else.

Reply:
I left a more in-depth analysis of why on the previous comment, it's basically because the types weren't actually used any and it was only serving as an artificially limited UI piece

Reply:
You didn't hit submit on the last comment (or at least it didn't show up).

Reply:
Dammit, github’s lying to me again. It was artificially constrained and required code changes for something that’s entirely a UI issue, I made the templates use the actual mime types to choose file icons for the source view. Behavior hasn’t changed on the Rust side, if anything it’s less complex, it just enables the frontend/templates to do more

Reply:
Oh I see, previously this was only used to choose which icon to display? In that case this is fine, yeah.